domain/secret/service/service.go

```diff
@@ -167,7 +71,7 @@ func (s *SecretService) CreateSecretURIs(ctx context.Context, count int) ([]*sec
 		return nil, errors.NotValidf("secret URi count %d", count)
 	}
 
-	modelUUID, err := s.st.GetModelUUID(ctx)
+	modelUUID, err := s.secretState.GetModelUUID(ctx)
```

Comment:
Here is the model uuid being fetched. I am not clear why we need it as an attribute...